homeassistant/components/blastbot_cloud/__init__.py

```diff
@@ -0,0 +1,88 @@
+"""The Blastbot Cloud integration."""
+import asyncio
+
+from blastbot_cloud_api.api import BlastbotCloudAPI
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
+
+from .const import DOMAIN, LOGGER
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+# List the platforms that you want to support.
+# Available: "switch", "climate", "remote"
+PLATFORMS = ["climate"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Blastbot Cloud component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Blastbot Cloud from a config entry."""
+    # Assign configuration variables.
+    # The configuration check takes care they are present.
+    username = entry.data.get(CONF_USERNAME)
+    password = entry.data.get(CONF_PASSWORD)
+
+    # Setup connection with devices/cloud
+    api = BlastbotCloudAPI()
+    successful_login = await api.async_login(username, password)
+
+    if not successful_login:
+        LOGGER.error("Could not connect to Blastbot Cloud")
+        await api.async_close()
+        return
```

Comment:
```suggestion
        raise ConfigEntryNotReady
```

Reply:
Can we distinguish bad credentials from bad connection? On recoverable errors like internet down, we want to raise ConfigEntryNotReady as above. On bad credentials we just want to log error and fail the setup by returning `False`.